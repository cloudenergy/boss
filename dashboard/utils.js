module.exports = {
  currency: (input)=> {
    return (input/100).toFixed(2)
  }
}
