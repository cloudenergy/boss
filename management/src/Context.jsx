import React from 'react'
import { Subject } from 'rxjs-compat'

export const ActionContext = React.createContext(new Subject());
