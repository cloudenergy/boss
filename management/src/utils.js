import {API_URI} from './env'
import * as r from 'ramda'
import {Observable} from 'rxjs-compat'

export function rest(url, opts) {
  let config = r.merge({url: `${API_URI}/v1.0/${url}`,
           crossDomain:true,
           withCredentials: true}, opts)
  return Observable
    .ajax(config)
    .catch(err => {
      switch (err.status) {
      case 401:
        window.location = "/login"
        break;
      default: throw err
      }
    })
}

export function modal(name, status) {
  window.$ && window.$(name).modal(status)
}
