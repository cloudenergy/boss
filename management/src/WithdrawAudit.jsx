import React from 'react'
import {Observable} from 'rxjs-compat'
import {rest} from './utils'
import * as r from 'ramda'
import './Finance.css'
import fuseOptFrom from './fuseOpt'
import Fuse from 'fuse.js'
import datef from 'dateformat'
import {AuditAction} from './Action'
import {AuditContext} from './Context'
import Confirm from './audit/Confirm'
import Table from './audit/Table'
const {Context, Var} = AuditContext

const fuseOpt = fuseOptFrom(['channel.project.name', 'channel.name', 'createdAt'])

const statusMap = {
  'PASSED': {color: 'text-success', text: '通过'},
  'AUDITFAILURE': {color: 'text-danger', text: '审核失败'},
  'PROCESSFAILURE': {color: 'text-danger', text: '处理失败'},
  'PENDING': {color: 'text-warning', text:'待审核'},
  'PROCESSING':{color: 'text-info', text: '正在处理'},
  'DONE': {color: 'text-success', text: '交易成功'}
}
const lensStatusMap = (status, val) => r.lensPath([status, val])

const withDrawTable =[{
  name: '项目名称',
  lens: r.view(r.lensPath(['channel', 'project', 'name']))
},{
  name: '交易账户',
  lens: r.view(r.lensPath(['channel', 'name']))
},{
  name: '提现金额',
  lens: r.view(r.lensProp('amount'))
},{
  name: '提现服务费',
  lens: r.view(r.lensPath(['']))
},{
  name: '到账金额',
  lens: r.view(r.lensProp(['meh']))
},{
  name: '操作后余额',
  lens: r.view(r.lensPath(['fundChannel', 'createdAt']))
},{
  name: '状态',
  lens: r.compose(status=><span className={r.view(lensStatusMap(status, 'color'))(statusMap)}>
    {r.view(lensStatusMap(status, 'text'))(statusMap)}
  </span>,
                  r.prop('status'))
},{
  name: '提交时间',
  lens: r.compose(x=>x && datef(Date.parse(x), 'yyyy年mm月dd日 HH:MM'),
                  r.view(r.lensPath(['createdAt'])))
}]

const contextValue = {
  actions: Var,
  table:withDrawTable,
  idLens: r.prop('id'),
  statusLens: r.compose(r.equals("PENDING"), r.prop('status')),
  modalId: "auditing",
  color: "bg-warning",
}

export default class WithdrawAudit extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      withDraw: [],
      auditId: '',
      auditEnable: true,
      query: "",
      fuse: new Fuse([], fuseOpt)
    }
  }
  componentDidMount() {
    this.subscription = Var.startWith(AuditAction.Load).flatMap(action => action.case({
      Load: () => rest('boss/finance')
        .map(({response})=>this.setState({
          withDraw: response.withDraw,
          fuse: new Fuse(response.withDraw, fuseOpt)
        })),
      Popup: (id,status) => Observable.of(this.setState({auditId: id, auditEnable: status})),
      Query: (str) => Observable.of(this.setState({query: str})),
      Approve: id => rest(`boss/withdraw/${id}/status`, {
        method: 'PUT',
        body: {status: 'PASSED'},
      }).flatMap(()=>Var.next(AuditAction.Load)),
      Deny: id => rest(`boss/withdraw/${id}/status`, {
        method: 'PUT',
        body: {status: 'AUDITFAILURE'},
      }).flatMap(()=>Var.next(AuditAction.Load))
    })).subscribe()
  }
  componentWillUnmount(){
    this.subscription.unsubscribe()
  }
  render() {
    let filtered = this.state.query? this.state.fuse.search(this.state.query): this.state.withDraw
    let selected = filtered.find(p=> r.prop('id')(p) === this.state.auditId )
    return (
      <Context.Provider value={contextValue}>
        <Confirm enable={this.state.auditEnable} data={selected} auditId={this.state.auditId} />
        <Table title="提现审核" data={filtered} />
      </Context.Provider>
    )
  }
}
