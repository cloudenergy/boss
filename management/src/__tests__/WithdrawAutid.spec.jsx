import React from 'react';
import renderer from 'react-test-renderer';
import * as r from 'ramda'
import Enzyme, {mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import {AuditContext,createActionContext} from '../Context'
import {Observable} from 'rxjs-compat'
import {rest} from '../utils.js'
jest.mock('../utils')
import Fuse from 'fuse.js'
import WithdrawAudit from '../WithdrawAudit';
import View from '../WithdrawAudit.view.jsx'
import reducer from '../WithdrawAudit.reducer.jsx'
const {Context, Var} = AuditContext
Enzyme.configure({ adapter: new Adapter() })

const data = {withDraw:[
  {
    "id": 2,
    "projectId": "6367598515924897792",
    "amount": 1000,
    "fundChannelId": 10,
    "operator": 15,
    "auditor": 0,
    "status": "DONE",
    "createdAt": "2018-03-19T07:12:37.000Z",
    "updatedAt": "2018-03-19T07:12:37.000Z",
    "deletedAt": null,
    "channel": {
      "id": 10,
      "flow": "pay",
      "projectId": "6367598515924897792",
      "category": "online",
      "tag": "alipay",
      "name": "张",
      "status": "PASSED",
      "createdAt": "2018-03-19T06:33:17.000Z",
      "updatedAt": "2018-07-15T10:44:35.000Z",
      "deletedAt": null,
      "project": {
        "id": "6367598515924897792",
        "logoUrl": null,
        "name": "公寓",
        "address": null,
        "description": null,
        "telephone": null,
        "createdAt": "2017-06-08T02:49:51.000Z",
        "updatedAt": "2017-06-08T02:49:51.000Z",
        "deletedAt": null
      }
    }
  },
  {
    "id": 3,
    "projectId": "6367598515924897792",
    "amount": 1000,
    "fundChannelId": 10,
    "operator": 15,
    "auditor": 0,
    "status": "PROCESSING",
    "createdAt": "2018-07-19T07:12:40.000Z",
    "updatedAt": "2018-09-19T07:12:40.000Z",
    "deletedAt": null,
    "channel": {
      "id": 10,
      "flow": "pay",
      "projectId": "6367598515924897792",
      "category": "online",
      "tag": "alipay",
      "name": "赵",
      "status": "PENDING",
      "createdAt": "2018-03-19T06:33:17.000Z",
      "updatedAt": "2018-07-15T10:44:35.000Z",
      "deletedAt": null,
      "project": {
        "id": "6367598515924897792",
        "logoUrl": null,
        "name": "公寓",
        "address": null,
        "description": null,
        "telephone": null,
        "createdAt": "2017-07-08T02:49:51.000Z",
        "updatedAt": "2017-06-08T02:49:51.000Z",
        "deletedAt": null
      }
    }
  }]}
rest.mockReturnValue(Observable.of({response:data}))
describe('<WithdrawAudit/>', ()=>{
  let subject, state= {
    withDraw: data.withDraw,
    auditId: '',
    fund: {},
    user: {},
    filters:{
      from: '2018-07-01',
      to: '2018-07-28',
      status: ''
    },
    summary: {
      sum: 0,
      withdraw: 0,
      fee:0,
      balance:0,
    },
    auditEnable: true,
    query: "",
    fuse: new Fuse([], {})
  }

  beforeEach(()=>{

  })

  it('renders <View />', () => {
    subject = (<View {...state} />)
    expect(renderer.create(subject).toJSON()).toMatchSnapshot()
  })

  it('renders <WithDrawAudit />', ()=> {
    subject = (<WithdrawAudit/>)
    expect(renderer.create(subject).toJSON()).toMatchSnapshot()
  })

  it('trigger search', (done) => {
    let setState = jest.fn()
    reducer(setState, state).subscribe(x=>{
      console.log(x)
      done()
    },err=>console.log(err))
  })
})
