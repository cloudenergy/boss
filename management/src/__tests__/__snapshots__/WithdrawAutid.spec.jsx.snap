// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithdrawAudit/> renders <View /> 1`] = `
Array [
  <div
    aria-hidden="true"
    className="modal"
    id="auditing"
    role="dialog"
    tabIndex="-1"
>
    <div
        className="modal-dialog modal-lg"
        role="document"
    >
        <div
            className="modal-content"
        >
            <div
                className="modal-header"
            >
                <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    type="button"
                >
                    <span
                        aria-hidden="true"
                    >
                        ×
                    </span>
                </button>
            </div>
            <div
                className="modal-body container"
            >
                <div
                    className="row"
                >
                    <div
                        className="col-3"
                    >
                        <img
                            alt="logo"
                            src="logo.png"
                            style={
                                Object {
                                    "width": "160px",
                                  }
                            }
                        />
                    </div>
                    <div
                        className="col-9 border-left"
                    >
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                项目名称
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                交易账户
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                提现金额
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                提交时间
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                交易状态
                                :
                            </dt>
                            <dd
                                className="col-8"
                            >
                                <span
                                    className={undefined}
                                />
                            </dd>
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                审核人
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                平台剩余金额
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                账户余额
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                    </div>
                </div>
            </div>
            <div
                className="modal-footer"
            >
                <button
                    className="btn btn-success"
                    data-dismiss="modal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                >
                    确认审核
                </button>
                <button
                    className="btn btn-danger"
                    data-dismiss="modal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                >
                    审核失败
                </button>
            </div>
        </div>
    </div>
</div>,
  <div
    className="accordion"
    id="banking-audit"
>
    <div
        className="card"
    >
        <div
            className="card-header bg-info"
        >
            <h5
                className="mb-0"
            >
                提现审核
            </h5>
        </div>
        <div>
            <div
                className="card-body"
            >
                <div
                    className="form-group row"
                >
                    <input
                        aria-label="Search"
                        className="form-control col-2"
                        onChange={[Function]}
                        placeholder="搜索"
                        type="search"
                    />
                    <div
                        className="col-6"
                    >
                        <input
                            className="col-3"
                            defaultValue="2018-07-01"
                            name="from"
                            onChange={[Function]}
                            required={true}
                            type="date"
                        />
                        <input
                            className="col-3"
                            defaultValue="2018-07-28"
                            name="to"
                            onChange={[Function]}
                            required={true}
                            type="date"
                        />
                        <select
                            className="col-3"
                            onChange={[Function]}
                        >
                            <option
                                value=""
                            >
                                全部状态
                            </option>
                            <option
                                value="PENDING"
                            >
                                待审核
                            </option>
                            <option
                                value="PROCESSFAILURE"
                            >
                                处理失败
                            </option>
                            <option
                                value="DONE"
                            >
                                交易成功
                            </option>
                        </select>
                    </div>
                    <div
                        className="col-2"
                    >
                        平台总金额: 
                        <span
                            className="text-warning"
                        />
                        元
                    </div>
                    <div
                        className="col-2"
                    >
                        可提现总金额: 
                        <span
                            className="text-success"
                        />
                        元
                    </div>
                </div>
                <table
                    className="table table-hover"
                >
                    <thead
                        className="sticky-top"
                    >
                        <tr>
                            <th
                                scope="col"
                            >
                                项目名称
                            </th>
                            <th
                                scope="col"
                            >
                                交易账户
                            </th>
                            <th
                                scope="col"
                            >
                                提现金额
                            </th>
                            <th
                                scope="col"
                            >
                                提交时间
                            </th>
                            <th
                                scope="col"
                            >
                                交易状态
                            </th>
                            <th
                                scope="col"
                            >
                                审核人
                            </th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr
                            data-toggle="modal"
                            onClick={[Function]}
                        >
                            <td>
                                公寓
                            </td>
                            <td>
                                赵
                            </td>
                            <td />
                            <td>
                                2018年07月19日 17:12
                            </td>
                            <td>
                                <span
                                    className={undefined}
                                />
                            </td>
                            <td />
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
    </div>
</div>,
]
`;

exports[`<WithdrawAudit/> renders <WithDrawAudit /> 1`] = `
Array [
  <div
    aria-hidden="true"
    className="modal"
    id="auditing"
    role="dialog"
    tabIndex="-1"
>
    <div
        className="modal-dialog modal-lg"
        role="document"
    >
        <div
            className="modal-content"
        >
            <div
                className="modal-header"
            >
                <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    type="button"
                >
                    <span
                        aria-hidden="true"
                    >
                        ×
                    </span>
                </button>
            </div>
            <div
                className="modal-body container"
            >
                <div
                    className="row"
                >
                    <div
                        className="col-3"
                    >
                        <img
                            alt="logo"
                            src="logo.png"
                            style={
                                Object {
                                    "width": "160px",
                                  }
                            }
                        />
                    </div>
                    <div
                        className="col-9 border-left"
                    >
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                项目名称
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                交易账户
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                提现金额
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                提交时间
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                交易状态
                                :
                            </dt>
                            <dd
                                className="col-8"
                            >
                                <span
                                    className={undefined}
                                />
                            </dd>
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                审核人
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                平台剩余金额
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                        <dl
                            className="row"
                        >
                            <dt
                                className="col-4"
                            >
                                账户余额
                                :
                            </dt>
                            <dd
                                className="col-8"
                            />
                        </dl>
                    </div>
                </div>
            </div>
            <div
                className="modal-footer"
            >
                <button
                    className="btn btn-success"
                    data-dismiss="modal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                >
                    确认审核
                </button>
                <button
                    className="btn btn-danger"
                    data-dismiss="modal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                >
                    审核失败
                </button>
            </div>
        </div>
    </div>
</div>,
  <div
    className="accordion"
    id="banking-audit"
>
    <div
        className="card"
    >
        <div
            className="card-header bg-info"
        >
            <h5
                className="mb-0"
            >
                提现审核
            </h5>
        </div>
        <div>
            <div
                className="card-body"
            >
                <div
                    className="form-group row"
                >
                    <input
                        aria-label="Search"
                        className="form-control col-2"
                        onChange={[Function]}
                        placeholder="搜索"
                        type="search"
                    />
                    <div
                        className="col-6"
                    >
                        <input
                            className="col-3"
                            defaultValue="2018-07-01"
                            name="from"
                            onChange={[Function]}
                            required={true}
                            type="date"
                        />
                        <input
                            className="col-3"
                            defaultValue="2018-07-31"
                            name="to"
                            onChange={[Function]}
                            required={true}
                            type="date"
                        />
                        <select
                            className="col-3"
                            onChange={[Function]}
                        >
                            <option
                                value=""
                            >
                                全部状态
                            </option>
                            <option
                                value="PENDING"
                            >
                                待审核
                            </option>
                            <option
                                value="PROCESSFAILURE"
                            >
                                处理失败
                            </option>
                            <option
                                value="DONE"
                            >
                                交易成功
                            </option>
                        </select>
                    </div>
                    <div
                        className="col-2"
                    >
                        平台总金额: 
                        <span
                            className="text-warning"
                        />
                        元
                    </div>
                    <div
                        className="col-2"
                    >
                        可提现总金额: 
                        <span
                            className="text-success"
                        />
                        元
                    </div>
                </div>
                <table
                    className="table table-hover"
                >
                    <thead
                        className="sticky-top"
                    >
                        <tr>
                            <th
                                scope="col"
                            >
                                项目名称
                            </th>
                            <th
                                scope="col"
                            >
                                交易账户
                            </th>
                            <th
                                scope="col"
                            >
                                提现金额
                            </th>
                            <th
                                scope="col"
                            >
                                提交时间
                            </th>
                            <th
                                scope="col"
                            >
                                交易状态
                            </th>
                            <th
                                scope="col"
                            >
                                审核人
                            </th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr
                            data-toggle="modal"
                            onClick={[Function]}
                        >
                            <td>
                                公寓
                            </td>
                            <td>
                                赵
                            </td>
                            <td />
                            <td>
                                2018年07月19日 17:12
                            </td>
                            <td>
                                <span
                                    className={undefined}
                                />
                            </td>
                            <td />
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
    </div>
</div>,
]
`;
